src/librustc/ich/impls_hir.rs

```diff
@@ -40,40 +40,6 @@ impl<'ctx> rustc_hir::HashStableContext for StableHashingContext<'ctx> {
         }
     }
 
-    // The following implementations of HashStable for `ItemId`, `TraitItemId`, and
-    // `ImplItemId` deserve special attention. Normally we do not hash `NodeId`s within
-    // the HIR, since they just signify a HIR nodes own path. But `ItemId` et al
-    // are used when another item in the HIR is *referenced* and we certainly
-    // want to pick up on a reference changing its target, so we hash the NodeIds
-    // in "DefPath Mode".
```

Comment:
These types do have different impls than `HirId`. Why can this now be changed?

Reply:
Seeing that comment I thought about how much the `NodeId`s have become deprecated after the lowering and wondered if we could get away with it. If there's a reason we should stick to this solution, though, we can probably still merge all these 3 types into 1, because their impls are identical.

Reply:
Yes, I think we should either merge them, or keep them separate if there's a good reason for them to remain separate. I'm not confident about the reason for their being separate in the first place, though, so I'd prefer if someone else clarified this.

Reply:
cc @michaelwoerister who [implemented this originally](https://github.com/rust-lang/rust/commit/c47cdc0d93726429ee18b418b1f85fae67b82d41).

Reply:
Merging the 3 types into 1 should be fine. Using an alias would be a change in semantics, leading to things being ignored during hashing.

Reply:
Any suggestion on the name? Would `ItemId` be fine?